microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,241 @@
+    exit_value: int = 0
+
+
+class Kselftest(Tool):
+    repo = "git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git"
+    mariner_repo = "https://github.com/microsoft/CBL-Mariner-Linux-Kernel"
+    # kselftest result log has "ok" and "not ok" prefixes, use regex to filter them
+    # example: not ok 1 selftests: breakpoints: step_after_suspend_test # exit=126
+    _RESULT_KSELFTEST_OK_REGEX = re.compile(r"ok (.*)\s+")
+
+    @property
+    def command(self) -> str:
+        return str(self._command)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def __init__(self, node: Node, kselftest_file_path: str, *args: Any, **kwargs: Any) -> None:
+        super().__init__(node, *args, **kwargs)
+        # command to run kselftests
+        self._command = (
+            self.get_tool_path(use_global=True)
+            / "kselftest-packages/run_kselftest.sh"
+        )
+
+        # tar file path specified in yml
+        self._tar_file_path = kselftest_file_path
+
+        if not self._tar_file_path:
+            # clone kernel, build kernel, then build kselftests
+            uname = self.node.tools[Uname]
+            uname_result = uname.get_linux_information(force_run=False)
+            version=uname_result.kernel_version
+            git = self.node.tools[Git]
+
+            if (version.patch == 0):
+                # clone major.minor kernel from upstream
+                branch = "linux-"
+                branch += str(version.major) + "." + str(version.minor) + ".y"
+                branch_to_clone = self.repo + " -b " + branch + " --depth 1"
+                kernel_path = git.clone(branch_to_clone, self.get_tool_path(use_global=True), fail_on_exists=False)
+                self.node.execute(cmd="make defconfig", cwd=kernel_path)
+            else:
+                if isinstance(self.node.os, CBLMariner):
+                    branch = "rolling-lts/mariner-2/"
+                    branch += str(version.major) + "." + str(version.minor) + "." + str(version.patch) + ".1"
+                    branch_to_clone = self.mariner_repo + " -b " + branch + " --depth 1"
+                    kernel_path = git.clone(branch_to_clone, self.get_tool_path(use_global=True), fail_on_exists=False)
+                    self.node.execute(cmd="wget 'https://raw.githubusercontent.com/microsoft/CBL-Mariner/2.0/SPECS/kernel/config' -O .config", cwd=kernel_path)
+                else:
+                    # clone major.minor.patch upstream stable kernel for distros other than Mariner
+                    branch += str(version.major) + "." + str(version.minor) + ".y"
+                    branch_to_clone = self.repo + " -b " + branch + " --depth 1"
+                    kernel_path = git.clone(branch_to_clone, self.get_tool_path(use_global=True), fail_on_exists=False)
+                    self.node.execute(cmd="make defconfig", cwd=kernel_path)
+            self.node.execute(cmd="make -j2", cwd=kernel_path)
```

Comment:
You can speed this up by using the number of cores this VM is running on. Please use `cores = node.tools[Lscpu].get_core_count()`

Reply:
> You can speed this up by using the number of cores this VM is running on. Please use `cores = node.tools[Lscpu].get_core_count()`

Updated code to use get_core_count().